ui/app/src/model/datasource-api.ts

```diff
@@ -11,40 +11,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { ProjectDatasource, DatasourceSelector, GlobalDatasource, Datasource } from '@perses-dev/core';
+import { ProjectDatasource, DatasourceSelector, GlobalDatasource } from '@perses-dev/core';
 import { DatasourceApi } from '@perses-dev/dashboards';
 import LRUCache from 'lru-cache';
 import { fetchDatasourceList } from './datasource-client';
 import { fetchGlobalDatasourceList } from './global-datasource-client';
 
-//TODO: Move to tanstack query for caching
 export class HTTPDatasourceAPI implements DatasourceApi {
-  getDatasource(
-    project: string,
-    selector: DatasourceSelector
-  ): Promise<{ resource: ProjectDatasource; proxyUrl: string } | undefined> {
+  /**
+   * Helper function for getting a proxy URL from separate input parameters.
+   * Give the following output according to the definition or not of the input.
+   * - /proxy/globaldatasources/{name}
+   * - /proxy/projects/{project}/datasources/{name}
+   * - /proxy/projects/{project}/dashboards/{dashboard}/{name}
+   *
+   * NB: despite the fact it's possible, it is useless to give a dashboard without a project as the url will for sure
+   * correspond to nothing.
+   * @param props.name
+   * @param props.dashboard
+   * @param props.project
+   */
+  buildProxyUrl(props: { project?: string; dashboard?: string; name: string }): string {
```

Comment:
this method is still a utils method and as such doesn't need to be attached to a class.

Reply:
I thought so but I thought of embeded use case, if the build of proxy url was part of the implementation of DatasourceApi before, then it should remain the same for local dashboards as well. Otherwise somebody integrating a embedded dashboard will be forced to use the proxy url format of the backend

Reply:
well if you are linked this method to the interface, then you are inducing the usage of proxy when it's not necessary the case depending of your usage.

You shouldn't provide or induce how the proxy url should be built. So for me it is still an utils method.

Reply:
I understand, but it was linked with the interface, I didn't change that. And actually I understand why it was linked with the interface.(Cf my previous answer)
I just moved it into a dedicated method instead of the return type. I can maybe set the method optional so that somebody implementing the DatasourseApi interface doesn't have to implement the buildProxyUrl. I will try.

Reply:
Let me give you an example with something you know well. If I develop a panel for the Argos UI, I will want the proxy url to be "/proxy/prometheus". I don´t want it to be computed according to an assomption of my backend. This was already possible before my implementation because the get methods of DatasourceApi was returning the proxyurl of each found datasource. But now, as we need also the datasource from the dashboard itself, we need something. 

I confess it's not ideal to expose all of it to somebody who want just an embedded panel, but well it's the way it's done as far as I understand :/

Reply:
yeah ok.

I re-read the code and ok you just change the way the `proxyURL` was exposed.

@jgbernalp This PR will break the datasource selector. I don't know exactly where are you in the integration of Perses but perhaps you would like to look at this PR. Perhaps you will have a different opinion about how the proxyURL should be exposed (like before, like it is proposed in this PR through a method in the interface, or in a different way)

Reply:
@Nexucis we are using the direct_url field on the datasource to fetch data from a service present inside a k8s cluster. So I have no strong opinion about the proxyURL.

Reply:
@jgbernalp I'm thinking of having the method that build the proxy url as optional so you could even not have to return a fake "" (as you might do currently)

Reply:
It seems more clear the new approach, to return what is being requested, rather than a wrapper object like before `{ resource: GlobalDataSource ...`.  I'm not against having the buildProxyUrl into a new function, but since this is not a react component maybe we can remove the `props` and just pass a plain parameter list.

Reply:
Yeah indeed I prefer the returns like that. 

Regarding the props, actually I like the way the props are passed in component and I wanted on purpose to do the same here as I have some optional/not optional parameters. 

First, we don´t have to guess the order of parameters, have project and/or dashboard undefined can be cumbersome sometimes. 
E.g if you want to have order name, project, dashboard and you don't want to define project, you have to put 
"mydatasource", undefined, "mydashboard"  

Second, we have named parameters which is quite easier to read. 

I can rename props to something else if you prefer like params

Reply:
So far I think the original thread can be resolved, moving the build of proxy url to a dedicated method seems to be ok. 

I mark this issue as resolved